map_it_test.go

```diff
@@ -1071,6 +1063,145 @@ func TestMap_ExecuteOnEntries(t *testing.T) {
 	})
 }
 
+func TestMap_AddInterceptor(t *testing.T) {
+	it.MapTester(t, func(t *testing.T, m *hz.Map) {
+		ctx := context.Background()
+		prefix := "My Prefix"
+		id, err := m.AddInterceptor(ctx, &MapGetInterceptor{Prefix: prefix})
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer func() {
+			if _, err := m.RemoveInterceptor(ctx, id); err != nil {
+				log.Fatal(err)
+			}
+		}()
+		v, err := m.Get(ctx, "foo")
+		if err != nil {
+			t.Fatal(err)
+		}
+		assert.Equal(t, prefix, v)
+		ok, err := m.RemoveInterceptor(ctx, id)
+		if err != nil {
+			log.Fatal(err)
+		}
+		assert.Equal(t, ok, true)
+		v, err = m.Get(ctx, "foo")
+		if err != nil {
+			t.Fatal(err)
+		}
+		assert.Equal(t, nil, v)
+	})
+}
+
+func TestMap_TryPut(t *testing.T) {
+	it.MapTester(t, func(t *testing.T, m *hz.Map) {
+		ctx1 := m.NewLockContext(context.Background())
+		if err := m.Lock(ctx1, "foo"); err != nil {
+			t.Fatal(err)
+		}
+		// TryPut with a different lock context returns false
+		ctx2 := m.NewLockContext(context.Background())
+		ok, err := m.TryPut(ctx2, "foo", "bar")
+		if err != nil {
+			t.Fatal(err)
+		}
+		assert.Equal(t, false, ok)
+		// TryPut with the same lock context returns true
+		ok, err = m.TryPut(ctx1, "foo", "bar")
+		if err != nil {
+			t.Fatal(err)
+		}
+		assert.Equal(t, true, ok)
+	})
+}
+
+func TestMap_TryPutWithTimeout(t *testing.T) {
+	it.MapTester(t, func(t *testing.T, m *hz.Map) {
+		ctx1 := m.NewLockContext(context.Background())
+		if err := m.Lock(ctx1, "foo"); err != nil {
+			t.Fatal(err)
+		}
+		// unlock 5 seconds later
+		go func() {
+			time.Sleep(5 * time.Second)
+			if err := m.Unlock(ctx1, "foo"); err != nil {
+				panic(err)
+			}
+		}()
+		// TryPut with a different lock context returns false
+		ctx2 := m.NewLockContext(context.Background())
+		ok, err := m.TryPutWithTimeout(ctx2, "foo", "bar", 1*time.Second)
```

Comment:
We can start the goroutine after this first failed put attempt to make sure it will never run before this. Sleep is not enough to make sure this test will not fail randomly.

Reply:
Updated at: https://github.com/hazelcast/hazelcast-go-client/pull/648/commits/7c277e2402b453759283fb1c77ceca950daad082